commons/src/main/java/com/powsybl/commons/datasource/TarArchiveDataSource.java

```diff
@@ -0,0 +1,329 @@
+        }
+
+        private static OutputStream getTmpStream(Path tarFilePath) throws IOException {
+            return new BufferedOutputStream(Files.newOutputStream(tarFilePath));
+        }
+
+        private static Path getTmpStreamFilePath(Path tarFilePath) {
+            return tarFilePath.getParent().resolve("tmp_stream_" + tarFilePath.getFileName() + ".stream");
+        }
+
+        private static TarArchiveOutputStream getTarStream(Path tmpTarFilePath) throws IOException {
+            return new TarArchiveOutputStream(new BufferedOutputStream(Files.newOutputStream(tmpTarFilePath)));
+        }
+
+        private static Path getTmpTarFilePath(Path tarFilePath) {
+            return tarFilePath.getParent().resolve("tmp_" + tarFilePath.getFileName());
+        }
+
+        private static Path getTmpCompressedTarFilePath(Path tarFilePath) {
+            return tarFilePath.getParent().resolve("tmp_comp_" + tarFilePath.getFileName());
+        }
+
+        private void compressTarFile() throws IOException {
+            try (InputStream fis = Files.newInputStream(getTmpTarFilePath(tarFilePath));
+                 OutputStream fos = Files.newOutputStream(getTmpCompressedTarFilePath(tarFilePath), StandardOpenOption.CREATE);
+                 OutputStream compressedOS = getCompressedOutputStream(fos, this.compressionFormat)) {
+                byte[] buffer = new byte[8192];
+                int len;
+                while ((len = fis.read(buffer)) != -1) {
+                    compressedOS.write(buffer, 0, len);
+                }
+            }
+        }
+
+        private static OutputStream getCompressedOutputStream(OutputStream os, CompressionFormat compressionFormat) throws IOException {
+            return compressionFormat == null ? os : switch (compressionFormat) {
+                case GZIP -> new GzipCompressorOutputStream(os);
+                case BZIP2 -> new BZip2CompressorOutputStream(os);
+                case XZ -> new XZCompressorOutputStream(os);
+                case ZSTD -> new ZstdCompressorOutputStream(os);
+                default -> os;
+            };
+        }
+
+        @Override
+        public void close() throws IOException {
+            if (!closed) {
+
+                // Close temporary stream file
+                super.close();
+
+                // Open a new temporary archive
+                try (TarArchiveOutputStream taos = getTarStream(getTmpTarFilePath(tarFilePath))) {
+
+                    // Useful parameter
+                    taos.setBigNumberMode(TarArchiveOutputStream.BIGNUMBER_POSIX);
+                    taos.setLongFileMode(TarArchiveOutputStream.LONGFILE_POSIX);
+
+                    // Copy content of temporary stream file into an entry of the temporary archive
+                    try (InputStream is = Files.newInputStream(getTmpStreamFilePath(tarFilePath))) {
+
+                        // Content of the stream
+                        byte[] streamContent = is.readAllBytes();
```

Comment:
isn't this a bit dangerous? That's only to get the size, couldn't you get it from `Files.size()`?

Reply:
not sure if it's dangerous but I can change it

Reply:
the problem is if the file is huge, readAllBytes will require a huge byte array, which could be avoided in case of a copy. And if the file is bigger than  `Integer.MAX_NUMBER`, it will be truncated.